xbmc/cores/AudioEngine/Sinks/AESinkDirectSound.cpp

```diff
@@ -451,122 +424,41 @@ double CAESinkDirectSound::GetCacheTotal()
 
 void CAESinkDirectSound::EnumerateDevicesEx(AEDeviceInfoList &deviceInfoList, bool force)
 {
-  CAEDeviceInfo        deviceInfo;
+  CAEDeviceInfo deviceInfo{};
 
-  ComPtr<IMMDeviceEnumerator> pEnumerator = nullptr;
-  ComPtr<IMMDeviceCollection> pEnumDevices = nullptr;
-  UINT uiCount = 0;
-
-  HRESULT                hr;
-
-  std::string strDD = GetDefaultDevice();
-
-  /* Windows Vista or later - supporting WASAPI device probing */
-  hr = CoCreateInstance(CLSID_MMDeviceEnumerator, NULL, CLSCTX_ALL, IID_IMMDeviceEnumerator, reinterpret_cast<void**>(pEnumerator.GetAddressOf()));
-  EXIT_ON_FAILURE(hr, "Could not allocate WASAPI device enumerator.")
-
-  hr = pEnumerator->EnumAudioEndpoints(eRender, DEVICE_STATE_ACTIVE, pEnumDevices.GetAddressOf());
-  EXIT_ON_FAILURE(hr, "Retrieval of audio endpoint enumeration failed.")
-
-  hr = pEnumDevices->GetCount(&uiCount);
-  EXIT_ON_FAILURE(hr, "Retrieval of audio endpoint count failed.")
-
-  for (UINT i = 0; i < uiCount; i++)
+  for (RendererDetail& detail : CAESinkFactoryWin::GetRendererDetails())
   {
-    ComPtr<IMMDevice> pDevice = nullptr;
-    ComPtr<IPropertyStore> pProperty = nullptr;
-    PROPVARIANT varName;
-    PropVariantInit(&varName);
-
     deviceInfo.m_channels.Reset();
     deviceInfo.m_dataFormats.clear();
     deviceInfo.m_sampleRates.clear();
+    deviceInfo.m_streamTypes.clear();
 
-    hr = pEnumDevices->Item(i, pDevice.GetAddressOf());
-    if (FAILED(hr))
-    {
-      CLog::LogF(LOGERROR, "Retrieval of DirectSound endpoint failed.");
-      goto failed;
-    }
-
-    hr = pDevice->OpenPropertyStore(STGM_READ, pProperty.GetAddressOf());
-    if (FAILED(hr))
+    if (detail.nChannels)
+      deviceInfo.m_channels =
+          layoutsByChCount[std::max(std::min(detail.nChannels, DS_SPEAKER_COUNT), 2U)];
+    deviceInfo.m_dataFormats.push_back(AEDataFormat(AE_FMT_FLOAT));
+    if (detail.eDeviceType != AE_DEVTYPE_PCM)
```

Comment:
Blank line to separate  if (detail.eDeviceType != AE_DEVTYPE_PCM) bloc

Also blank line after end of bloc.

Reply:
sure.